### ISSUE TITLE
[WASM] `ComboBox` scrolling to the bottom and re-opening is very slow

### ISSUE THREAD

**Issue by kevin100702** [unoplatform/uno#9775]
### Current behavior

My ComboBox has around 300 Items and when I last make a selection scroll the scrollbar to the bottom, select the last element, and when I pull down the selection again it takes a long time to open. But if I last selected the element above, it's very quick to open again.

### Expected behavior

It can also be opened quickly when the scroll bar is scrolled to the bottom.

### Environment

Uno.UI / Uno.UI.WebAssembly / Uno.UI.Skia

### NuGet package version(s)

4.4.2

### Affected platforms

WebAssembly

### IDE

Visual Studio 2022

### IDE version

17.3.3

**Comment by MartinZikmund** [comment]
I am guessing this is related to #556

**Comment by jeromelaban** [comment]
@MartinZikmund not necessarily. It may be related to the scrolling and materialization of previous items on explicit item selection, as Wasm already has virtualization in place for ComboBox.

**Comment by kevin100702** [comment]
According to some of my tests, it actually has something to do with the last scrollbar position. When I scroll the scroll bar to the bottom even with no item selected, it's slow the next time I open it. I tried to clear the SelectedIndex and SelectedItem after selected the item, and it is still very slow the next time I open it.

**Comment by TimGels** [comment]
@kevin100702 
> According to some of my tests, it actually has something to do with the last scrollbar position. When I scroll the scroll bar to the bottom even with no item selected, it's slow the next time I open it. I tried to clear the SelectedIndex and SelectedItem after selected the item, and it is still very slow the next time I open it.

I can confirm this behavior as well using Uno 6 with Skia. But it actually is a memory leak issue when I keep an eye on the memory usage.

![Image](https://github.com/user-attachments/assets/76c693e2-b952-46a3-aa1e-da014dea87d9)

**Comment by MartinZikmund** [comment]
# Repro generator

Runtime test(s) added in branch [`dev/mazi/repro/9775`](https://github.com/unoplatform/uno/compare/dev/mazi/repro/9775):
- `src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_ComboBox.cs` → `When_ComboBox_With_Many_Items_Reopens_After_Scrolling_To_Bottom`
- `src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_ComboBox.cs` → `When_ComboBox_With_Many_Items_Reopens_After_Selecting_First_Vs_Last`

⚠️ The test(s) **pass on Skia targets**. The issue reports the bug on **WebAssembly**, which was using native DOM rendering at the time of filing (v4.4.2, September 2022). Since then, WASM has migrated to Skia-based rendering, which may have resolved this performance issue. The tests verify that reopening a 300-item ComboBox after selecting the last item completes within a reasonable time (<3s) and is not dramatically slower than after selecting the first item. Both assertions pass comfortably on Skia Desktop.

The branch is provided as a starting point — if the issue persists on current WASM, the test logic should help narrow down the performance bottleneck.
